aten/src/ATen/native/cuda/ReduceMomentKernel.cu

```diff
@@ -36,26 +36,40 @@ static void std_var_kernel_cuda(TensorIterator& iter, bool unbiased, bool take_s
 
 template <typename scalar_t, typename acc_t=scalar_t, typename out_t=scalar_t>
```

Comment:
setting `acc_t = typename c10::scalar_value_type<scalar_t>::type` should resolve the issue here

`c10::scalar_value_type<scalar_t>::type` returns `scalar_t` for all non-complex dtypes and returns `T` for `c10::complex<T>`.

Reply:
Thanks so much for the tip. The latest code now manipulates `c10::scalar_value_type<acc_t>::type` to get the type of the factor, the overload functions for complex numbers are not needed.